Library/src/Systems/Settings/SettingsSystem.cpp

```diff
@@ -868,4 +872,66 @@ void SettingsSystem::RemoveAvatarPortrait(const csp::common::String& UserId, Nul
 
 	GetAvatarPortraitAssetCollection(UserId, PortraitAvatarAssetCollCallback);
 }
+
+void SettingsSystem::SetAvatarInfo(const csp::common::String& InUserId,
+								   AvatarType InType,
+								   const csp::common::Variant& InIdentifier,
+								   NullResultCallback Callback)
+{
+	rapidjson::Document Json;
+	Json.SetObject();
+	Json.AddMember("type", static_cast<int>(InType), Json.GetAllocator());
+	Json.AddMember("identifierType", static_cast<int>(InIdentifier.GetValueType()), Json.GetAllocator());
+
+	switch (InIdentifier.GetValueType())
+	{
+		case csp::common::VariantType::Integer:
+			Json.AddMember("identifier", InIdentifier.GetInt(), Json.GetAllocator());
+			break;
+		case csp::common::VariantType::String:
+			Json.AddMember("identifier", rapidjson::Value(InIdentifier.GetString().c_str(), InIdentifier.GetString().Length()), Json.GetAllocator());
+			break;
+		default:
+			throw std::exception("Unsupported Avatar Identifier type");
```

Comment:
CSP doesn't currently follow a pattern of throwing exceptions when passed unsupported types. 

Since the current pattern is to print to the Error log and early out (random example [here](https://github.com/magnopus-opensource/connected-spaces-platform/blob/develop/Library/src/Systems/Users/UserSystem.cpp#L270)), could we change it follow the same pattern instead?

Reply:
Address in latest commit